lib/hooks.d.ts

```diff
@@ -1,11 +1,43 @@
-export declare type UsePromiseParams = {
-    initialValue?: any;
-    reducer?: Function;
-};
+export declare type UsePromiseResult<T> = T | unknown;
+export declare type UsePromiseCall<T> = Promise<T | Error>;
+export declare type UsePromiseError = Error | null;
+export interface UsePromiseOptions<T, U> {
+    onCompleted?: (result: T) => void;
+    onError?: (error: Error) => void;
+    reducer?: (state: T) => U;
+    skip?: boolean;
+    initialState?: U;
+}
+export interface UsePromiseRef<T, U> extends UsePromiseOptions<T, U> {
+    promise: () => Promise<T>;
+    mounted: boolean;
+}
+export interface UsePromiseState<T> {
+    loading: boolean;
+    result: UsePromiseResult<T>;
+    error: UsePromiseError;
```

Comment:
```suggestion
    error: Error | null;
```

aqui creo que no hace falta una interfaz para eso